doc/theming.rst

```diff
@@ -310,6 +290,44 @@ Python :mod:`ConfigParser` module) and has the following structure:
   and are accessible from all templates as ``theme_<name>``.
 
 
+.. _distribute-your-theme:
+
+Distribute your theme as a python package
+-----------------------------------------
+
+As a way to distribute your theme, you can use python package.  Python package
+brings to users easy setting up ways.
+
+To distribute your theme as a python package, please define an entry point
+called ``sphinx.html_themes`` in your setup.py file, and write a ``setup()``
+function to register your themes in it::
+
+    # 'setup.py'
+    setup(
+        ...
+        entry_points = {
+            'sphinx.html_themes': [
+                'name_of_theme = your_package',
+            ]
+        },
+        ...
+    )
+
+    # 'your_package.py'
+    from os import path
+
+    def setup(app):
+        app.add_html_theme('name_of_theme', path.abspath(path.dirname(__file__)))
```

Comment:
From theme-extension's sight, the name parameter is not needed. The name is already defined in entry point. So this parameter is duplicated. In addition, I think almost theme package might be contains only one theme. A package having multiple themes might be rare.

On the other hand, from API's sight, the name is important. Our theme manager looks up a theme using its name. So it is hard to look up if we drop the parameter.